File: examples/ios/CobolDemo/CobolDemo/bridge/cobol-to-ios/iOSBridge.m
#import <AVFoundation/AVFoundation.h>

/**
 * Expose the speech synthesizer api in a single objective C function.
 */
void speakObjC(const char *message) {
    // Keep the synthesizer around for later invocations.
    static AVSpeechSynthesizer *synthesizer = nil;
    if (!synthesizer) synthesizer = [[AVSpeechSynthesizer alloc] init];

    NSString *str = [NSString stringWithUTF8String:message];
    AVSpeechUtterance *utterance = [AVSpeechUtterance speechUtteranceWithString:str];
    [synthesizer speakUtterance:utterance];
}

File: examples/ios/CobolDemo/CobolDemo/bridge/cobol-to-ios/iOSBridge.swift
import AVFoundation
private let synthesizer = AVSpeechSynthesizer()

/**
 * Expose the speech synthesizer api in a single swift function.
 */
@_cdecl("speakSwift")
func speakSwift(message: UnsafePointer<CChar>) {
    // https://developer.apple.com/documentation/avfoundation/speech-synthesis
    let utterance = AVSpeechUtterance(string: String(cString: message))
    synthesizer.speak(utterance)
}
